File: qc-application/src/main/java/com/eager/questioncloud/application/api/authentication/dto/SocialAuthenticationResult.java
package com.eager.questioncloud.application.api.authentication.dto;

import lombok.AllArgsConstructor;
import lombok.Getter;

@Getter
@AllArgsConstructor
public class SocialAuthenticationResult {
    private Boolean isRegistered;
    private String registerToken;
    private AuthenticationToken authenticationToken;

    public static SocialAuthenticationResult success(AuthenticationToken authenticationToken) {
        return new SocialAuthenticationResult(true, null, authenticationToken);
    }

    public static SocialAuthenticationResult notRegister(String registerToken) {
        return new SocialAuthenticationResult(false, registerToken, null);
    }
}

File: qc-application/src/main/kotlin/com/eager/questioncloud/application/api/authentication/dto/SocialAuthenticationResult.kt
package com.eager.questioncloud.application.api.authentication.dto

class SocialAuthenticationResult(
    val isRegistered: Boolean,
    val registerToken: String?,
    val authenticationToken: AuthenticationToken?,
) {
    companion object {
        fun success(authenticationToken: AuthenticationToken?): SocialAuthenticationResult {
            return SocialAuthenticationResult(true, null, authenticationToken)
        }

        fun notRegister(registerToken: String?): SocialAuthenticationResult {
            return SocialAuthenticationResult(false, registerToken, null)
        }
    }
}
